OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1044,8 +1044,43 @@ internal static class DreamProcNativeRoot {
                     return new DreamValue(list);
                 }
                 case JsonValueKind.Object: {
+                    // Stick to using an enumerator because getting an array, specially of an
+                    // object in a plaintext notation of unconstrained size, causes lot of
+                    // heap allocation, so avoid that.
+                    var enumerator = jsonElement.EnumerateObject();
                     DreamList list = objectTree.CreateList();
 
+                    // The object contained nothing.
+                    if(!enumerator.MoveNext())
+                        return new DreamValue(list);
+
+                    // For handling special values expressed as single-property objects
+                    // Such as float-point values Infinity and NaN
+                    var first = enumerator.Current;
+                    var has_second = enumerator.MoveNext();
+                    if (!has_second) {
```

Comment:
This should be renamed to `hasSecond`
```suggestion
                    var hasSecond = enumerator.MoveNext();
                    if (!hasSecond) {
```